Panel.php

```diff
@@ -137,6 +137,9 @@ public function getTraceLine($options)
             return $options['text'];
         }
 
+        if ($this->module->traceBasePath && is_string($this->module->traceBasePath)) { 
```

Comment:
> is_string($this->module->traceBasePath)
I think is redundant condition, if you are going to use it only as string then better bring it to string in setTraceBasePath function.

Reply:
Fair enough. Which do you think I should do? Setter function or just remove the condition? I just wanted to be strict about typing there, because as-is it doesn't support it being a callable/array/etc

Reply:
i think setter function clearer

Reply:
Like this?

```php
    /**
     * Sets the base path for trace lines to replace the application's base path
     * 
     * @param string $basePath the base path
     */
    public function setTraceBasePath($basePath)
    {
        if (!is_string($basePath)) {
            return;
        }
        $this->_traceBasePath = $basePath;
    }
```

and changed the property to `public $_traceBasePath;`

Reply:
```php
private function _traceBasePath
public function setTraceBasePath($basePath)
{
    $this->_traceBasePath = (string) $basePath;
}
public function getTraceBasePath()
{
    return $this->getTraceBasePath()
}
```

and add phpdoc to class
```
@property $traceBasePath string ...
```

Reply:
@dmirogin done, thanks for the tips :smile: